rmgpy/molecule/group.py

```diff
@@ -1274,7 +1521,27 @@ def isIdentical(self, other):
             return False
         else:
             return True
+        
+    def labeledIsSubgraphIsomorphic(self,other):
+        """
+        Check that two labeled groups have a subgraph isomorphism that matches the labels
+        Note that this implementation requires two subgraph isomorphism operations
+        When the atoms being mapped are known to be subgraph isomorphic (such as in most RMG operations) 
+        one should use isSubgraphIsomorphic with the initialMap input instead
+        """
+        cython.declare(a=GroupAtom,L=list,initialMap=dict)
 
+        initialMap = dict()
+        for atom in self.atoms:
+            if atom.label and atom.label != '':
+                L = [a for a in other.atoms if a.label == atom.label]
+                assert len(L) == 1, 'more than one label for {0} in {1}'.format(atom.label,other)
+                initialMap[atom] = L[0]
+        
+        #isSubgraphIsomorphic ignores the atoms in the initial map so we have to do it seperately without
+        #the map to confirm that the atoms in the initial map are subgraph-isomorphic
+        return self.isSubgraphIsomorphic(other,initialMap=initialMap) and self.isSubgraphIsomorphic(other) #necessary
```

Comment:
I'm not sure if this will always hold true. You're better off confirming that the initial map is valid using `isMappingValid` then running the subgraph isomorphism check. I also think that you could simply add automatic generation of the initial map as an option to the existing `isSubgraphIsomorphic` method instead of creating a new method.

Same applies for the method in `Molecule`.

Reply:
I made a lot of changes related to subgraphIsomorphism checking in the not yet a PR part 2 of automatic tree generation.  Since we don't intend to use part 1 without part 2 (I broke up the PRs to make it easier to review) is it alright if I make this change in part 2?

Reply:
this should be fixed now